Dockerfile

```diff
@@ -24,10 +24,13 @@ RUN mv /app/config/database.yml.example /app/config/database.yml
 
 
 RUN gem install bundler io-console --no-ri --no-rdoc && \
-  bundle config set without 'development test' && \
+  bundle config set --local without 'development test' && \
   bundle install --jobs 20 --retry 5
 
-RUN RAILS_ENV=production RAILS_GROUPS=js SECRET_KEY_BASE=1234 bin/rails assets:precompile
+RUN RAILS_ENV=production RAILS_GROUPS=assets SECRET_KEY_BASE=1234 bin/rails assets:precompile
+
+RUN bundle config set --local without 'development test assets' && \
```

Comment:
Can this instead just be? :

    RUN BUNDLE_WITHOUT="assets" bundle clean --force`

None of the development/test gems are installed so setting the full `bundle config..` again seems a bit unnecessary.

Reply:
Hi, I tried this, didn't work for me. `bundle config set` by default sets local config and [local config takes precedence over the env](https://github.com/rubygems/rubygems/blob/a753447de8fb92dad30e927656984aa2b7765f3d/bundler/lib/bundler/settings.rb#L86-L87). I tried `--global` as well but didn't work either. Maybe you can give this a try?

> so setting the full bundle config.. again seems a bit unnecessary.

internally both ` bundle config set without` and `BUNDLE_WITHOUT` set up the value of the same setting `Bundler.settings['without']`. Perhaps using the same method to change the same setting make more sense?

Reply:
I am moving ahead with this. Please free to update this if you can find a way do this without using `bundle config set`